packages/app/src/device-file-system/commands/retrieve-files.command.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { Endpoint, Method } from "@mudita/pure"
+import { AppError } from "App/core/errors"
+import { Result, ResultObject } from "App/core/builder"
+import { RequestResponseStatus } from "App/core/types/request-response.interface"
+import { BaseCommand } from "App/device-file-system/commands/base.command"
+import { DeviceFileSystemError } from "App/device-file-system/constants"
+
+export class RetrieveFilesCommand extends BaseCommand {
+  public async exec(directory: string): Promise<
+    ResultObject<
+      | Record<
+          string,
+          {
+            path: string
+            fileSize: number
+          }[]
+        >
+      | undefined
+    >
+  > {
```

Comment:
It might be a personal preference, but I would find something like this to be more readable:
```
type RetrievedFile = {
  path: string
  fileSize: number
}
...
Promise<FailedResult | SuccessResult<Record<string, RetrievedFile>>>
```
(but it might require some changes in `result.builder.ts` file. Feel free to ignore if you disagree